server/src/test/java/org/opensearch/action/admin/cluster/stats/ClusterStatsRequestBuilderTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.stats;
+
+import org.opensearch.action.admin.cluster.stats.ClusterStatsRequest.IndexMetric;
+import org.opensearch.action.admin.cluster.stats.ClusterStatsRequest.Metric;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.client.NoOpClient;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.Set;
+
+public class ClusterStatsRequestBuilderTests extends OpenSearchTestCase {
+
+    private NoOpClient testClient;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        this.testClient = new NoOpClient(getTestName());
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        this.testClient.close();
+        super.tearDown();
+    }
+
+    public void testUseAggregatedNodeLevelResponses() {
+        ClusterStatsRequestBuilder clusterStatsRequestBuilder = new ClusterStatsRequestBuilder(
+            this.testClient,
+            ClusterStatsAction.INSTANCE
+        );
+        clusterStatsRequestBuilder.useAggregatedNodeLevelResponses(false);
+        assertFalse(clusterStatsRequestBuilder.request().useAggregatedNodeLevelResponses());
```

Comment:
Can we serialize and de-serialize the request before asserting for the values ?

Reply:
Serde test cases for ClusterStatsRequest are written separately, I don't think we need to cover serde here.

Reply:
Ok. Do you think we can combine with them by constructing the request via builder ? Not seeing much value in having a test-class for validation of setters.

Reply:
Makes sense, it should be okay to delete the builder test file.